scene/gui/control.cpp

```diff
@@ -1419,13 +1419,29 @@ void Control::_set_global_position(const Point2 &p_point) {
 
 void Control::set_global_position(const Point2 &p_point, bool p_keep_offsets) {
 	ERR_MAIN_THREAD_GUARD;
+
+	if (p_point == get_global_position()) {
+		return; // Edge case, but avoids calculation.
+	}
+
 	Transform2D inv;
 
 	if (data.parent_canvas_item) {
 		inv = data.parent_canvas_item->get_global_transform().affine_inverse();
 	}
 
-	set_position(inv.xform(p_point), p_keep_offsets);
+	Point2 new_position;
+	if (data.pivot_offset.is_zero_approx() || (data.scale.is_equal_approx(Vector2(1.0, 1.0)) && Math::is_zero_approx(Math::fmod(data.rotation, (real_t)Math_TAU)))) {
```

Comment:
BTW such check for the rotation would fail in case it's almost `TAU` etc. like `359.999999` degrees (it could be close to no rotation but not near-zero after `fmod`ding it).

Alternative could be e.g. `wrap`ping it into `[-PI, PI]` instead (and then checking only for near-zero). Or just adding here an additional check for near-`TAU`. Not sure if/what is preferred.